sorc/chgres_cube.fd/input_data.F90

```diff
@@ -5604,6 +5950,180 @@ SUBROUTINE READ_FV3_GRID_DATA_NETCDF(FIELD,TILE_NUM,IMO,JMO,LMO, &
  ERROR = NF90_CLOSE(NCID)
 
  END SUBROUTINE READ_FV3_GRID_DATA_NETCDF
+ 
+ !---------------------------------------------------------------------------
+! Read winds from a grib2 file
+!---------------------------------------------------------------------------
+
+ subroutine read_winds(file,inv,u,v,localpet)
+
+ use wgrib2api
+ use netcdf
+ use program_setup, only      : get_var_cond, fix_dir_input_grid
+ use model_grid, only         : input_grid_type
+ implicit none
+
+ character(len=250), intent(in)          :: file
+ character(len=10), intent(in)            :: inv
+ integer, intent(in)                     :: localpet
+ real(esmf_kind_r8), intent(inout), allocatable :: u(:,:,:),v(:,:,:)
+
+ real(esmf_kind_r4), dimension(i_input,j_input)  :: alpha
+ real(esmf_kind_r8), dimension(i_input,j_input)  :: lon, lat
+ real(esmf_kind_r4), allocatable                 :: u_tmp(:,:),v_tmp(:,:)
+ real(esmf_kind_r4), dimension(i_input,j_input)  :: ws,wd
+ real(esmf_kind_r4)                      :: value_u, value_v,lov,latin1,latin2
+ real(esmf_kind_r8)                      :: d2r
+
+ integer                                 :: varnum_u, varnum_v, vlev, & !ncid, id_var, &
+                                            error, iret, i,istr
+
+ character(len=20)                       :: vname
+ character(len=50)                       :: method_u, method_v
+ character(len=250)                      :: file_coord, cmdline_msg
+ character(len=10000)                    :: temp_msg
+
+ d2r=acos(-1.0_esmf_kind_r8) / 180.0_esmf_kind_r8
+ if (localpet==0) then
+   allocate(u(i_input,j_input,lev_input))
+   allocate(v(i_input,j_input,lev_input))
+ else
+   allocate(u(0,0,0))
+   allocate(v(0,0,0))
+ endif
+
+ file_coord = trim(fix_dir_input_grid)//"/latlon_grid3.32769.nc"
+ 
+ vname = "u"
+ call get_var_cond(vname,this_miss_var_method=method_u, this_miss_var_value=value_u, &
+                       loc=varnum_u)
+ vname = "v"
+ call get_var_cond(vname,this_miss_var_method=method_v, this_miss_var_value=value_v, &
+                       loc=varnum_v)
+
+ if (trim(input_grid_type)=="rotated_latlon") then
+   print*,"- CALL FieldGather FOR INPUT GRID LONGITUDE"
+   call ESMF_FieldGather(longitude_input_grid, lon, rootPet=0, tile=1, rc=error)
+   if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+        call error_handler("IN FieldGather", error)
+   print*,"- CALL FieldGather FOR INPUT GRID LATITUDE"
+   call ESMF_FieldGather(latitude_input_grid, lat, rootPet=0, tile=1, rc=error)
+   if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+        call error_handler("IN FieldGather", error)
+
+   if (localpet==0) then
+     print*,"- CALCULATE ROTATION ANGLE FOR ROTATED_LATLON INPUT GRID"
+     cmdline_msg = "wgrib2 "//trim(file)//" -d 1 -grid &> temp.out"
+     call system(cmdline_msg)
```

Comment:
These system calls are kind of clunky.  Do they work on all platforms?  There may be a cleaner way to return the map information.

Reply:
They should work on all platforms, yes. The path to the wgrib2 executable is included in the path by default in the release configuration for all primary supported platforms. I'm open to other suggestions if we can find another way to get this information.

Reply:
So wgrib2 will not retrieve information about the map projection (pass it back in memory)?  Maybe Wesley E. knows of a way to do that (I can ask him).  Also, the NCEP grib2 library will not only degrib, but can pass back all header information (the grid description template, etc).  That library is an option to retrieve the header info you need.